docs: october release notes (#8957)
adobe/react-spectrum: packages/dev/docs/pages/releases/2025-10-02.mdx

{/* Copyright 2025 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {BlogPostLayout, Hero, Image} from '@react-spectrum/docs';
export default BlogPostLayout;

---

> description: Happy fall! This release brings some long-awaited features, including multi-selection in React Aria Select and enhanced animation support across our components. We've also addressed key bugs, such as better collection handling with React Suspense, and improved mobile behavior in React Aria Modal, especially for iOS 26, with fixes for pinch zooming and text selection. Lastly, `onAction` can now be used on a `ComboBoxItem` to perform a custom action when the item is selected, enabling the ability to add a “Create” option for the current input value.

Reviewer: ```suggestion
description: Happy fall! This release brings some long-awaited features, including multi-selection in React Aria Select and enhanced animation support across our components. For example, Disclosure now provides CSS variables to easily animate expanding and collapsing. We’ve also added a new <SelectionIndicator> component that can be used within collection items to animate selection changes. In addition, we have significantly improved the behavior of scrolling modals on iOS 26. Lastly, onAction can now be used on a ComboBox item, enabling you to add a “create” option.
```

date: 2025-10-02
---

# October 2, 2025 Release

> Happy fall! This release brings some long-awaited features, including multi-selection in [React Aria Select]() and enhanced animation support across our components. For example, [React Aria Disclosure](../react-aria/Disclosure.html) now provides `—disclosure-panel-width` and `—disclosure-panel-height` CSS variables to help you implement smooth animations. We've also added a new `SelectionIndicator` component that can be used within collection items (e.g., [GridList](../react-aria/GridList.html#anatomy), [ListBox](../react-aria/ListBox.html#anatomy), [Menu](../react-aria/Menu.html#anatomy), [RadioGroup](../react-aria/RadioGroup.html#anatomy), [Table](../react-aria/Table.html#anatomy), [Tabs](../react-aria/Tabs.html#example), [TagGroup](../react-aria/TagGroup.html#anatomy), [ToggleButtonGroup](../react-aria/ToggleButtonGroup.html#animation), [Tree](../react-aria/Tree.html#anatomy)) to animate selection changes.

Reviewer 2: ```suggestion
Happy fall! This release brings some long-awaited features, including multi-selection in [React Aria Select]() and enhanced animation support across our components. For example, [React Aria Disclosure](../react-aria/Disclosure.html) now provides `--disclosure-panel-width` and `--disclosure-panel-height` CSS variables to help you implement smooth animations. We've also added a new `SelectionIndicator` component that can be used within collection items (e.g., [GridList](../react-aria/GridList.html#anatomy), [ListBox](../react-aria/ListBox.html#anatomy), [Menu](../react-aria/Menu.html#anatomy), [RadioGroup](../react-aria/RadioGroup.html#anatomy), [Table](../react-aria/Table.html#anatomy), [Tabs](../react-aria/Tabs.html#example), [TagGroup](../react-aria/TagGroup.html#anatomy), [ToggleButtonGroup](../react-aria/ToggleButtonGroup.html#animation), [Tree](../react-aria/Tree.html#anatomy)) to animate selection changes.
```

Reviewer 2: > component that can be used within collection items

Any component with single selection, RadioGroup isn't a Collection component. Trying to think of a good way to write that

> Happy fall! This release brings some long-awaited features, including multi-selection in [React Aria Select]() and enhanced animation support across our components. For example, [React Aria Disclosure](../react-aria/Disclosure.html) now provides `—disclosure-panel-width` and `—disclosure-panel-height` CSS variables to help you implement smooth animations. We've also added a new `SelectionIndicator` component that can be used within collection items (e.g., [GridList](../react-aria/GridList.html#anatomy), [ListBox](../react-aria/ListBox.html#anatomy), [Menu](../react-aria/Menu.html#anatomy), [RadioGroup](../react-aria/RadioGroup.html#anatomy), [Table](../react-aria/Table.html#anatomy), [Tabs](../react-aria/Tabs.html#example), [TagGroup](../react-aria/TagGroup.html#anatomy), [ToggleButtonGroup](../react-aria/ToggleButtonGroup.html#animation), [Tree](../react-aria/Tree.html#anatomy)) to animate selection changes.

Reviewer 3: double check this one just in case, I don't have the build running right now
```suggestion
Happy fall! This release brings some long-awaited features, including multi-selection in [React Aria Select](../react-aria/Select.html#multiple-selection) and enhanced animation support across our components. For example, [React Aria Disclosure](../react-aria/Disclosure.html) now provides `—disclosure-panel-width` and `—disclosure-panel-height` CSS variables to help you implement smooth animations. We've also added a new `SelectionIndicator` component that can be used within collection items (e.g., [GridList](../react-aria/GridList.html#anatomy), [ListBox](../react-aria/ListBox.html#anatomy), [Menu](../react-aria/Menu.html#anatomy), [RadioGroup](../react-aria/RadioGroup.html#anatomy), [Table](../react-aria/Table.html#anatomy), [Tabs](../react-aria/Tabs.html#example), [TagGroup](../react-aria/TagGroup.html#anatomy), [ToggleButtonGroup](../react-aria/ToggleButtonGroup.html#animation), [Tree](../react-aria/Tree.html#anatomy)) to animate selection changes.
```

Reviewer 3: oh nevermind it matches the link further below so should be fine

> In addition, this release includes important bug fixes such as improved collection handling with React Suspense, as well as enhancements to React Aria Modal, particularly for those using iOS 26, which also includes fixing the pinch zooming and text selection on mobile. Lastly, `onAction` can now be used on a ComboBoxItem to perform a custom action when the item is selected, enabling the ability to add a “Create” option for the current input value.

Reviewer: ```suggestion
```

merged this paragraph into the previous one

### Multi-Selection in React Aria Select:

> In order to support multi-selection in React Aria Select, we are updating the API from `selectedKey` to  `value`. The deprecated API is supported for backward compatibility, but only supports single selection. Please see our [docs](../react-aria/Select.html#multiple-selection) to see more on how to implement multi-selection in Select. 

Reviewer: ```suggestion
To support multi-selection in React Aria Select, we are updating the API from `selectedKey` to `value`, matching the React `<select>` API. The deprecated API is supported for backward compatibility, but only supports single selection. Please see our [docs](../react-aria/Select.html#multiple-selection) for more on how to implement multi-selection.
```

* Disclosure
  * Add support for disclosure animation for React Aria Disclosure - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/8867)
* Select
  * Add support for multiple selection to React Aria Select - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/8734), [@lixiaoyan](https://github.com/lixiaoyan) - [PR](https://github.com/adobe/react-spectrum/pull/8880)
* Miscellaneous
  * Add support for animated selection indicators - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/8876)
* Modals
  * Improve behavior of modals on iOS 26 - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/8888), [@LFDanLu](https://github.com/LFDanLu) - [PR](https://github.com/adobe/react-spectrum/pull/8948)
  * Allow pinch zooming and text selection in Modals on iOS - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/8922)

## Fixes

* Color

>   * Fix ColorLoupe position in ColorSlider for RTL locales - [@chirokas](https://github.com/chirokas) - [PR](https://github.com/adobe/react-spectrum/pull/8894)

Author: oh oops, yes i can correct that
